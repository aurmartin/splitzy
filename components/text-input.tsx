import {
  TextInput as RNTextInput,
  View,
  type TextInputProps as RNTextInputProps,
  type StyleProp,
  type ViewStyle,
} from "react-native";
import { Text } from "./text";
import React from "react";

type TextInputProps = {
  label?: string;
  right?: React.ReactNode;
  style?: StyleProp<ViewStyle>;
  ref?: React.Ref<RNTextInput>;
} & RNTextInputProps;

function TextInput(props: TextInputProps) {
  const { ref, label, right, style, ...rest } = props;

  return (
    <View
      style={[
        {
          borderRadius: 8,
          paddingHorizontal: 16,
          backgroundColor: "#fff",
          flexDirection: "row",
          alignItems: "center",
          height: 55,
        },
        style,
      ]}
    >
      {label ? <Text>{label}</Text> : null}

      <View style={{ flex: 1, justifyContent: "center" }}>
        <RNTextInput
          {...rest}
          ref={ref}
          style={{
            flex: 1,
            textAlign: label ? "right" : "left",
            fontSize: 16,
            paddingTop: 0,
            paddingBottom: 0,
          }}
          placeholderTextColor="#888"
          accessibilityLabel={label}
        />
      </View>

      {right ? <View style={{ marginLeft: 2 }}>{right}</View> : null}
    </View>
  );
}

export { TextInput, type TextInputProps };
