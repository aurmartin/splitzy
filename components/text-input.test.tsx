import * as React from "react";
import { render, screen, userEvent } from "@testing-library/react-native";
import { TextInput } from "./text-input";

describe("TextInput", () => {
  it("renders correctly", () => {
    render(<TextInput placeholder="Enter text" />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("renders with label when provided", () => {
    render(<TextInput label="Test Label" placeholder="Enter text" />);
    screen.getByText("Test Label");
  });

  it("handles text changes correctly", async () => {
    const user = userEvent.setup();
    const mockOnChangeText = jest.fn();
    render(
      <TextInput placeholder="Enter text" onChangeText={mockOnChangeText} />,
    );

    const input = screen.getByPlaceholderText("Enter text");
    await user.type(input, "Hello World");

    expect(mockOnChangeText).toHaveBeenCalledWith("Hello World");
  });
});
