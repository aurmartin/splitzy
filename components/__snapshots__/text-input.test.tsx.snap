// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 55,
        "paddingHorizontal": 16,
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <TextInput
      placeholder="Enter text"
      placeholderTextColor="#888"
      style={
        {
          "flex": 1,
          "fontSize": 16,
          "paddingBottom": 0,
          "paddingTop": 0,
          "textAlign": "left",
        }
      }
    />
  </View>
</View>
`;
